components/kserve/kserve.go

```diff
@@ -166,6 +183,43 @@ func (k *Kserve) ReconcileComponent(ctx context.Context, cli client.Client, resC
 	return nil
 }
 
+func (k *Kserve) setDefaultDeploymentMode(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
+	inferenceServiceConfigMap := &corev1.ConfigMap{}
+	err := cli.Get(ctx, client.ObjectKey{
+		Namespace: dscispec.ApplicationsNamespace,
+		Name:      "inferenceservice-config",
+	}, inferenceServiceConfigMap)
+	if err != nil {
+		fmt.Println("error to get configmap 'inferenceservice-config'")
+		return err
+	}
+
+	//set data.deploy.defaultDeploymentMode to the model specified in the Kserve spec
+	var deployData map[string]interface{}
+	json.Unmarshal([]byte(inferenceServiceConfigMap.Data["deploy"]), &deployData)
+	deployData["defaultDeploymentMode"] = k.DefaultDeploymentMode
+	deployDataBytes, _ := json.MarshalIndent(deployData, "", " ")
+	inferenceServiceConfigMap.Data["deploy"] = string(deployDataBytes)
+
+	//if the default deployment mode is RawDeployment, also set the ingressClass to "openshift-default", otherwise it should be "istio"
+	var ingressData map[string]interface{}
+	json.Unmarshal([]byte(inferenceServiceConfigMap.Data["ingress"]), &ingressData)
+	if k.DefaultDeploymentMode == RawDeployment {
+		ingressData["ingressClassName"] = RawDeploymentIngressClassName
```

Comment:
Move to constant

Reply:
removed the logic for `IngressClassName` based on your PR to allow disableIngressCreation